lib/parsers/package-lock-parser.ts

```diff
@@ -119,7 +120,12 @@ export class PackageLockParser implements LockfileParser {
       } else if (/^file:/.test(dep.version)) {
         depTree.dependencies[dep.name] = createPkgTreeFromDep(dep);
       } else {
-        throw new OutOfSyncError(depName, 'npm');
+        if (!strictOutOfSync) {
```

Comment:
I would sugget to flip the direction, i.e.:
```
if (strictOutOfSync) {
  throw new OutOfSyncError(depName, 'npm');
}
...
```

Saving one `else` statement :-)